canvas_sdk/commands/tests/tests.py

```diff
@@ -456,3 +464,125 @@ def test_command_schema_matches_command_api(
         assert len(expected_field["choices"]) == len(choices)
         for choice in choices:
             assert choice["value"] in expected_field["choices"]
+
+
+@pytest.fixture
+def commands_to_test() -> list[Any]:
+    return [
+        AssessCommand,
+        DiagnoseCommand,
+        GoalCommand,
+        HistoryOfPresentIllnessCommand,
+        MedicationStatementCommand,
+        PlanCommand,
+        PrescribeCommand,
+        QuestionnaireCommand,
+        ReasonForVisitCommand,
+        StopMedicationCommand,
+        UpdateGoalCommand,
+    ]
```

Comment:
Same thing here -- the way this is being used is more akin to a parametrized test than a fixture. This is all static data, so there isn't really any setup required here, which is what fixtures are usually for (think issuing HTTP requests to create things). Given that it would probably be better to use pytest test parametrization to accomplish this.

Reply:
love the idea of making it parametrized but i use this list to also generate the protocol code. i'll see if there's a way to reuse it in that way

Reply:
Given that these are just in alphabetical order, I'm assuming that there are no dependencies between them like in the other case. This would actually be pretty simple. A parametrized fixture would look like this:

```python
# For reuse with the protocol code
COMMANDS = [
    AssessCommand,
    DiagnoseCommand,
    GoalCommand,
    HistoryOfPresentIllnessCommand,
    MedicationStatementCommand,
    PlanCommand,
    PrescribeCommand,
    QuestionnaireCommand,
    ReasonForVisitCommand,
    StopMedicationCommand,
    UpdateGoalCommand,
]

# Specifying "ids" means each scenario gets its own label
@pytest.fixture(params=COMMANDS, ids=[c.Meta.key for c in COMMANDS])
def commands_to_test(request: FixtureRequest) -> _BaseCommand:
    return request.param

```

And then you would pass in the fixture to the test, removing the need to loop over the  commands in the test.

Reply:
after looking into this more, i don't think parametrization makes sense. the goal is to create a test where a single protocol inserts a note with every command. so its really just one single test and one single protocol. if we went the parametrization route, we would be doing a different protocol for each command, which would take longer to test, and is unnecessary